src/kubernetes.cc

```diff
@@ -102,6 +109,68 @@ bool ReadServiceAccountSecret(
   return !input.fail();
 }
 
+// Extracts the API group and version from the api_version.
+const std::pair<std::string, std::string> ApiGroupAndVersion(
+    const std::string& api_version) {
+  std::vector<std::string> slash_split;
+  boost::algorithm::split(
+      slash_split, api_version, boost::algorithm::is_any_of("/"));
+  if (slash_split.size() == 1) {
+    // api_version has the format "<version>". E.g. "v1".
+    return std::make_pair(std::string(kK8sCoreGroup), slash_split[0]);
+  } else if (slash_split.size() == 2) {
+    // api_version has the format "<group-name>/<version>". E.g. "apps/v1".
+    return std::make_pair(slash_split[0], slash_split[1]);
+  } else {
+    LOG(ERROR) << "Invalid apiVersion: " << api_version;
+  }
+}
+
+// Builds the resource type and version of a Kubernetes resource given its API
+// version and kind.
+const std::pair<std::string, std::string> TypeAndVersion(
+    const std::string& api_version, const std::string& kind) {
+  const std::pair<std::string, std::string> group_and_version =
+      ApiGroupAndVersion(api_version);
+  const std::string group_name = group_and_version.first;
+  std::string resource_type;
+  if (group_name == kK8sCoreGroup) {
```

Comment:
This looks convoluted to me. You've invented `kK8sCoreGroup` for the sole purpose of comparing it here. Would have been simpler to do a `slash_split` test, the way you did in `FullResourceName`. Also in `WatchPath`, though see my other comment...